hphp/util/asm-x64.cpp

```diff
@@ -38,6 +38,11 @@ const char* show(RoundDirection rd) {
   not_reached();
 }
 
+#ifdef HAVE_LIBXED
+  static XedInit xi;
+  constexpr xed_state_t XedAssembler::s_xedState;
```

Comment:
I think you're relying on the initialization order here (ie, xed_state_t should be created after xed_tables_init, which is called by XedInit's constructor). I believe the initialization order is guaranteed in this case, but it's worth adding a comment here calling out this implicit dependence.

Reply:
There's no dependency between kXedState and the Xed initialization.  kXedState is just used later as a mandatory argument to the xed_inst* method.